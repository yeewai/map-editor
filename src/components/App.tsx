import * as React from 'react';
import { Route, Switch } from 'react-router-dom';

import StructuresView from './StructuresView';
import WorldsView from './WorldsView';

import { ModalRoot, ConfirmModal } from 'ever-modal';
import EditStructureDefinitionModal from './StructuresView/EditStructureDefinitionModal';
import AddStructureDefinitionModal from './StructuresView/AddStructureDefinitionModal';

class App extends React.Component {
    render() {
        const modal_components = {

            'ADD_STRUCTURE_DEFINITION': AddStructureDefinitionModal,
            'EDIT_STRUCTURE_DEFINITION': EditStructureDefinitionModal
        };

        return (
            <div className="App">
                <Switch>
                    <Route path="/structures" component={StructuresView} />
                    <Route path="/worlds" component={WorldsView} />
                </Switch>

                <ModalRoot modal_components={modal_components}/>
                <ModalRoot modal_components={{'CONFIRM_MODAL': ConfirmModal}}/>
            </div>
        );
    }
}

export default App;
