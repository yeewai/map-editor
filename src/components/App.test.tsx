import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './App';

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
    it ("matches snapshot", () => {
        const wrapper = shallow(<App/>);
        expect(wrapper).toMatchSnapshot();
    })
})
