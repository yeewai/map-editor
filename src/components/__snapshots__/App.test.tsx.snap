// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Switch>
          <Route
                    component={[Function]}
                    path="/structures"
          />
          <Route
                    component={[Function]}
                    path="/worlds"
          />
</Switch>,
        <Connect(ModalRoot)
          modal_components={
                    Object {
                              "ADD_STRUCTURE_DEFINITION": [Function],
                              "EDIT_STRUCTURE_DEFINITION": [Function],
                            }
          }
/>,
        <Connect(ModalRoot)
          modal_components={
                    Object {
                              "CONFIRM_MODAL": [Function],
                            }
          }
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/structures"
/>,
            <Route
              component={[Function]}
              path="/worlds"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/structures",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/worlds",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "modal_components": Object {
            "ADD_STRUCTURE_DEFINITION": [Function],
            "EDIT_STRUCTURE_DEFINITION": [Function],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "modal_components": Object {
            "CONFIRM_MODAL": [Function],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Switch>
            <Route
                        component={[Function]}
                        path="/structures"
            />
            <Route
                        component={[Function]}
                        path="/worlds"
            />
</Switch>,
          <Connect(ModalRoot)
            modal_components={
                        Object {
                                    "ADD_STRUCTURE_DEFINITION": [Function],
                                    "EDIT_STRUCTURE_DEFINITION": [Function],
                                  }
            }
/>,
          <Connect(ModalRoot)
            modal_components={
                        Object {
                                    "CONFIRM_MODAL": [Function],
                                  }
            }
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/structures"
/>,
              <Route
                component={[Function]}
                path="/worlds"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/structures",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/worlds",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "modal_components": Object {
              "ADD_STRUCTURE_DEFINITION": [Function],
              "EDIT_STRUCTURE_DEFINITION": [Function],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "modal_components": Object {
              "CONFIRM_MODAL": [Function],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
